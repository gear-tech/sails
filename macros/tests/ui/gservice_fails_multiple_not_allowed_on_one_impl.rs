use sails_macros::gservice;

struct MyService;

#[gservice]
#[gservice]
impl MyService {
    pub fn this(&self, p1: bool) -> bool {
        !p1
    }
}

#[tokio::main]
async fn main() {}
