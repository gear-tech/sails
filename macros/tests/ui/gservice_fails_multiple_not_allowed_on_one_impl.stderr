error: multiple `gservice` attributes on a type with the same name are not allowed
 --> tests/ui/gservice_fails_multiple_not_allowed_on_one_impl.rs:7:1
  |
7 | impl MyService {
  | ^^^^

error[E0599]: no method named `this` found for struct `MyService` in the current scope
 --> tests/ui/gservice_fails_multiple_not_allowed_on_one_impl.rs:8:12
  |
3 | struct MyService;
  | ---------------- method `this` not found for this struct
...
8 |     pub fn this(&self, p1: bool) -> bool {
  |            ^^^^ method not found in `MyService`
